examples/ethernet-rtic-stm32h735g-dk.rs

```diff
@@ -51,44 +47,40 @@ static mut DES_RING: ethernet::DesRing<4, 4> = ethernet::DesRing::new();
 
 /// Net storage with static initialisation - another global singleton
 pub struct NetStorageStatic<'a> {
-    ip_addrs: [IpCidr; 1],
     socket_storage: [SocketStorage<'a>; 8],
-    neighbor_cache_storage: [Option<(IpAddress, Neighbor)>; 8],
-    routes_storage: [Option<(IpCidr, Route)>; 1],
 }
 static mut STORE: NetStorageStatic = NetStorageStatic {
     // Garbage
-    ip_addrs: [IpCidr::Ipv6(Ipv6Cidr::SOLICITED_NODE_PREFIX)],
     socket_storage: [SocketStorage::EMPTY; 8],
-    neighbor_cache_storage: [None; 8],
-    routes_storage: [None; 1],
 };
 
 pub struct Net<'a> {
-    iface: Interface<'a, ethernet::EthernetDMA<'a, 4, 4>>,
+    iface: Interface,
+    ethdev: ethernet::EthernetDMA<4, 4>,
+    sockets: SocketSet<'a>,
 }
 impl<'a> Net<'a> {
     pub fn new(
-        store: &'static mut NetStorageStatic<'a>,
-        ethdev: ethernet::EthernetDMA<'a, 4, 4>,
+        store: &'a mut NetStorageStatic<'a>,
+        mut ethdev: ethernet::EthernetDMA<4, 4>,
         ethernet_addr: HardwareAddress,
     ) -> Self {
-        // Set IP address
-        store.ip_addrs = [IpCidr::new(IpAddress::v4(192, 168, 1, 99), 0)];
+        let mut config = Config::new();
+        config.hardware_addr = Some(ethernet_addr);
 
-        let neighbor_cache =
-            NeighborCache::new(&mut store.neighbor_cache_storage[..]);
-        let routes = Routes::new(&mut store.routes_storage[..]);
+        let mut iface = Interface::new(config, &mut ethdev);
+        // Set IP address
+        iface.update_ip_addrs(|addrs| {
+            let _ = addrs.push(IpCidr::new(IpAddress::v4(192, 168, 1, 99), 0));
+        });
 
-        let iface =
-            InterfaceBuilder::new(ethdev, &mut store.socket_storage[..])
-                .hardware_addr(ethernet_addr)
-                .neighbor_cache(neighbor_cache)
-                .ip_addrs(&mut store.ip_addrs[..])
-                .routes(routes)
-                .finalize();
+        let sockets = SocketSet::new(&mut store.socket_storage[..]);
 
-        Net { iface }
+        Net::<'a> {
```

Comment:
It's not actually necessary to annotate the lifetime `'a` here, but there's no harm to it (and it can stay!)